cmd/boostd/genindex.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	_ "net/http/pprof"
+	"os"
+	"time"
+
+	"github.com/filecoin-project/boost/cmd/lib"
+	"github.com/filecoin-project/boost/piecedirectory"
+	"github.com/filecoin-project/boostd-data/model"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/ipfs/go-cid"
+	carv1 "github.com/ipld/go-car"
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/multiformats/go-varint"
+	"github.com/urfave/cli/v2"
+)
+
+var genindexCmd = &cli.Command{
+	Name:  "genindex",
+	Usage: "Generate index for a given piececid and store in the piece directory",
+	Flags: []cli.Flag{
+		&cli.IntFlag{
+			Name:  "add-index-throttle",
+			Usage: "the maximum number of add index operations that can run in parallel",
+			Value: 4,
+		},
+		&cli.StringFlag{
+			Name:     "api-fullnode",
+			Usage:    "the endpoint for the full node API",
+			Required: false,
+		},
+		&cli.StringFlag{
+			Name:     "api-storage",
+			Usage:    "the endpoint for the storage node API",
+			Required: false,
+		},
+		&cli.StringFlag{
+			Name:     "api-piece-directory",
+			Usage:    "the endpoint for the piece directory API",
+			Required: false,
+		},
+		&cli.StringFlag{
+			Name:     "piececid",
+			Usage:    "piececid to index",
+			Required: false,
```

Comment:
Looking at the code in the Action it seems like piececid should be a required variable
Nit: suggest calling it piece-cid to match the casing of the other parameters.